gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherSettings.java

```diff
@@ -337,4 +346,66 @@ public Object extractNextToken(ListTopicSubscriptionsResponse payload) {
               return payload.getSubscriptionsList();
             }
           };
+
+  private static BundlingDescriptor<PublishRequest, PublishResponse> PUBLISH_BUNDLING_DESC =
+      new BundlingDescriptor<PublishRequest, PublishResponse>() {
+        @Override
+        public String getBundlePartitionKey(PublishRequest request) {
+          return request.getTopic();
+        }
+
+        @Override
+        public PublishRequest mergeRequests(Collection<PublishRequest> requests) {
+          PublishRequest firstRequest = requests.iterator().next();
+
+          List<PubsubMessage> elements = new ArrayList<>();
+          for (PublishRequest request : requests) {
+            elements.addAll(request.getMessagesList());
+          }
+
+          PublishRequest bundleRequest =
+              PublishRequest.newBuilder()
+                  .setTopic(firstRequest.getTopic())
+                  .addAllMessages(elements)
+                  .build();
+          return bundleRequest;
+        }
+
+        @Override
+        public void splitResponse(
+            PublishResponse bundleResponse,
+            Collection<? extends RequestIssuer<PublishRequest, PublishResponse>> bundle) {
+          int bundleMessageIndex = 0;
+          for (RequestIssuer<PublishRequest, PublishResponse> responder : bundle) {
+            List<String> subresponseElements = new ArrayList<>();
+            int subresponseCount = responder.getRequest().getMessagesCount();
+            for (int i = 0; i < subresponseCount; i++) {
```

Comment:
Any reason for not using for each loop? Is response.getRequest().getMessageCount() can be less
than the size of bundleResponse.getMessageIdsList()?

Also (nit pick), you can Initialize subresponseElements to the expected size if you already know it.